File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/observer/observer/ObserverTwo.java
package com.bigboxcode.designpattern.observer.observer;

public class ObserverTwo extends Observer{

    public ObserverTwo(Subject subject){
        this.subject = subject;
        this.subject.attach(this);
    }

    @Override
    public void sendUpdate() {
        System.out.println( "Received in ObserverTwo: " + subject.getState());
    }
}

File: design-patterns/php/observer/general/ObserverTwo.php
<?php
// ObserverTwo.php

namespace BigBoxCode\DesignPattern\Observer\General;

class ObserverTwo extends Observer {
    public function __construct(Subject $subject) {
        $this->subject = $subject;
        $this->subject->attach($this);
    }

    public function sendUpdate(): void {
        echo "Received in ObserverTwo: " . $this->subject?->getState() . "\n";
    }
}